src/QueueConnection.js

```diff
@@ -115,6 +116,25 @@ class QueueConnection {
     }
   }
 
+  /**
+   * @return Promise
+   * */
+  close (handleCloseEvent = false) {
+    if (this._connection) {
+      if (!handleCloseEvent) {
+        this._connection.off('close', this._onClose)
+      }
+
+      this._connection.close((err) => {
```

Comment:
Itt a promise API-t használd a close method-ból!
Egyszerűbb lesz kezelni a hibát.
Illetve ettől async lesz ez a function is, szóval be kell várni a promise-okat ahol ez hívva lesz.